b3sum/src/main.rs

```diff
@@ -496,26 +510,43 @@ fn parse_check_line(mut line: &str) -> Result<ParsedCheckLine> {
 }
 
 fn hash_one_input(path: &Path, args: &Args) -> Result<()> {
-    let mut input = Input::open(path, args)?;
-    let output = input.hash(args)?;
-    if args.raw() {
-        write_raw_output(output, args)?;
-        return Ok(());
-    }
-    if args.no_names() {
+    let md = metadata(path).unwrap();
+    if md.is_dir() && args.recurse() {
+        let mut entries = fs::read_dir(path)?
+            .map(|res| res.map(|e| e.path()))
+            .collect::<Result<Vec<_>, io::Error>>()?;
```

Comment:
`.collect::<Result<Vec<_>, io::Error>>()?` very smooth but now I'm not sure I believe you when you say "I'm a Rust newbie" :)

Reply:
I can neither confirm nor deny this was copied from [some documentation I looked up](https://doc.rust-lang.org/std/fs/fn.read_dir.html). 😁